src/main/java/com/spring/taskmanger/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,43 @@
+package com.spring.taskmanger.security;
+
+import com.spring.taskmanger.dao.UserRepository;
+import com.spring.taskmanger.exception.ResourceNotFoundException;
+import com.spring.taskmanger.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Override
+    @Transactional
+
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```

Comment:
I think the method name should be loadUserByUserEmail

Reply:
UserDetailsService interface has just this method that has to be implemented 
but inside this method i loaded the user by email 
        User user = userRepository.findByEmail(email)